app-modules/data-migration/database/migrations/2024_02_26_000000_create_one_time_operations_table.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+use AdvisingApp\DataMigration\OneTimeOperationManager;
+
+class CreateOneTimeOperationsTable extends Migration
+{
+    protected string $name;
+
+    public function __construct()
+    {
+        $this->name = OneTimeOperationManager::getTableName();
+    }
+
+    public function up(): void
+    {
+        Schema::create($this->name, function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('name');
+            $table->string('dispatched');
+            $table->timestamp('processed_at')->nullable();
```

Comment:
I'm wondering if we should make these timestamps a little bit more robust to account for failure and multiple process attempts. What do you think about adding `first_processed_at` and `last_processed_at`? For more granularity, we can also add a count for `process_attempts`.

Reply:
I think we would want to ensure these only ever run once. So, tracking that may not be needed. If the job that get's dispatch fails we can retry that job.

If we do ever need to re-run the command there is the option with this to pass the operation name directly. If you do so it runs it regardless of if it has run before.